metadata/metadata.go

```diff
@@ -1686,119 +1751,163 @@ func (serv *MetadataServer) CreateSourceVariant(ctx context.Context, variant *pb
-	return serv.genericList(ENTITY, func(msg proto.Message) error {
+	_, ctx, logger := serv.Logger.InitializeRequestID(stream.Context())
+	logger.Info("Opened List Entities stream")
+	return serv.genericList(ctx, ENTITY, func(msg proto.Message) error {
 		return stream.Send(msg.(*pb.Entity))
 	})
 }
 
-func (serv *MetadataServer) CreateEntity(ctx context.Context, entity *pb.Entity) (*pb.Empty, error) {
-	return serv.genericCreate(ctx, &entityResource{entity}, nil)
+func (serv *MetadataServer) CreateEntity(ctx context.Context, entityRequest *pb.EntityRequest) (*pb.Empty, error) {
+	logger := serv.Logger.WithRequestID(logging.RequestID(entityRequest.RequestId)).WithResource("entity", entityRequest.Entity.Name, "")
+	logger.Info("Creating Entity")
+	ctx = logging.UpdateContext(ctx, logger, entityRequest.RequestId)
+	return serv.genericCreate(ctx, &entityResource{entityRequest.Entity}, nil)
 }
 
 func (serv *MetadataServer) GetEntities(stream pb.Metadata_GetEntitiesServer) error {
-	return serv.genericGet(stream, ENTITY, func(msg proto.Message) error {
+	_, ctx, logger := serv.Logger.InitializeRequestID(stream.Context())
+	logger.Info("Opened Get Entities stream")
+	return serv.genericGet(ctx, stream, ENTITY, func(msg proto.Message) error {
 		return stream.Send(msg.(*pb.Entity))
 	})
 }
 
 func (serv *MetadataServer) ListModels(_ *pb.Empty, stream pb.Metadata_ListModelsServer) error {
-	return serv.genericList(MODEL, func(msg proto.Message) error {
+	_, ctx, logger := serv.Logger.InitializeRequestID(stream.Context())
+	logger.Info("Opened List Models stream")
+	return serv.genericList(ctx, MODEL, func(msg proto.Message) error {
 		return stream.Send(msg.(*pb.Model))
 	})
 }
 
-func (serv *MetadataServer) CreateModel(ctx context.Context, model *pb.Model) (*pb.Empty, error) {
-	return serv.genericCreate(ctx, &modelResource{model}, nil)
+func (serv *MetadataServer) CreateModel(ctx context.Context, modelRequest *pb.ModelRequest) (*pb.Empty, error) {
+	logger := serv.Logger.WithRequestID(logging.RequestID(modelRequest.RequestId)).WithResource("model", modelRequest.Model.Name, "")
+	logger.Info("Creating Model")
+	ctx = logging.UpdateContext(ctx, logger, modelRequest.RequestId)
+	return serv.genericCreate(ctx, &modelResource{modelRequest.Model}, nil)
 }
 
 func (serv *MetadataServer) GetModels(stream pb.Metadata_GetModelsServer) error {
-	return serv.genericGet(stream, MODEL, func(msg proto.Message) error {
+	_, ctx, logger := serv.Logger.InitializeRequestID(stream.Context())
+	logger.Info("Opened Get Models stream")
+	return serv.genericGet(ctx, stream, MODEL, func(msg proto.Message) error {
 		return stream.Send(msg.(*pb.Model))
 	})
 }
 
 // GetEquivalent attempts to find an equivalent resource based on the provided ResourceVariant.
 func (serv *MetadataServer) GetEquivalent(ctx context.Context, req *pb.ResourceVariant) (*pb.ResourceVariant, error) {
-	return serv.getEquivalent(req, true)
+	_, ctx, logger := serv.Logger.InitializeRequestID(ctx)
```

Comment:
You shouldn't be initializing a new request ID, if it comes from api/main.go it's being attached to the request and sent over. We should use that instead.